lib/text_parser.dart

```diff
@@ -25,27 +25,25 @@ List<TextSegment> parseText(String? text) {
 
     final url = match.namedGroup('url');
     final keyword = match.namedGroup('keyword');
-    final tag = match.namedGroup('tag');
 
     if (url != null) {
       segments.add(TextSegment(url, url, false, false, false, null, true));
     } else if (keyword != null) {
       final isWord = match.start == 0 ||
           [' ', '\n'].contains(text.substring(match.start - 1, start));
       if (!isWord) {
-        return;
+        continue;
       }
 
       final isHashtag = keyword.startsWith('#');
       final isMention = keyword.startsWith('@');
       final isTag = keyword.startsWith('<');
-      var subString;
+      late String subString;
```

Comment:
Add some documentation for Regix logic

Reply:
`
// if there is ' ' or '\n' before keyword i.e is there is blank space or \n before @mention, #hashtag etc e.g " @mention", "\n @mention"
      final isWord = match.start == 0 ||
          [' ', '\n'].contains(text.substring(match.start - 1, start));
      // if there is not ' ' or '\n' then continue the loop e.g in the case "sometext@mention"
      if (!isWord) {
        continue;
      }
`

Reply:
`
/// if keyword is custom tag e.g <tag>some text</tag>
      /// extract 'tag' and remove <tag>, </tag> from keyword using regex
      if (isTag) {
        final regexTag = RegExp(r"[<>]"); // <>
        final regexStartTag = RegExp(r"<\w+>"); // <tag>
        final regexEndTag = RegExp(r"<\/*\w*>*"); //</tag>
        tagName = keyword.replaceAll(regexTag, "");
        keywordToDisplay = keyword.replaceAll(regexStartTag, "");
        keywordToDisplay = keywordToDisplay.replaceAll(regexEndTag, "");
      } else {
        keywordToDisplay = keyword;
      }
`